arrow-core-test/src/main/kotlin/arrow/test/laws/MonadLogicLaws.kt

```diff
@@ -0,0 +1,135 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.core.Option
+import arrow.core.Tuple2
+import arrow.core.extensions.eq
+import arrow.core.extensions.option.eq.eq
+import arrow.core.extensions.tuple2.eq.eq
+import arrow.test.generators.GenK
+import arrow.test.generators.functionAToB
+import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
+import arrow.typeclasses.MonadLogic
+import arrow.typeclasses.reflect
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import io.kotlintest.shouldBe
+
+object MonadLogicLaws {
+
+  fun <F> laws(
+    ML: MonadLogic<F>,
+    GK: GenK<F>,
+    EQK: EqK<F>,
+    iterations: Int = 1000
```

Comment:
What is the idea behind this? This is too low for stacksafety and too high for normal testing isn't it?

Reply:
i had some issues with long running tests for StateT so i added an option to control the number of iterations. in my understanding 1000 here is the default used by kotlintest.

https://github.com/arrow-kt/arrow/pull/2047/files#diff-55e755bcfe1e88678c7c35d4aec3432bR93

tbh: i didn't understand what was the underlying reason for the slow tests ...